integration-test/src/test/java/org/jboss/pnc/integration_new/endpoint/BuildConfigurationEndpointTest.java

```diff
@@ -606,4 +613,151 @@ public void shouldUpdateBuildConfigurationWithDependencies() throws ClientExcept
         assertThat(updatedBC.getEnvironment().getId()).isEqualTo(environmentId);
         assertThat(modificationTime).isNotEqualTo(updatedBC.getModificationTime());
     }
+
+    @Test
+    @InSequence(61)
+    public void shouldChangeRepositoryConfiguration() throws ClientException {
+        // given
+        BuildConfigurationClient bcClient = new BuildConfigurationClient(RestClientConfiguration.asUser());
+        SCMRepositoryClient scmClient = new SCMRepositoryClient(RestClientConfiguration.asAnonymous());
+        BuildConfiguration buildConfiguration = bcClient.getSpecific(configuration4Id);
+
+        // make sure this RC is not already set
+        assertThat(buildConfiguration.getScmRepository().getId()).isNotEqualTo(repositoryConfiguration2Id);
+
+        // when
+        BuildConfiguration updateBC = BuildConfiguration.builder()
+                .id(buildConfiguration.getId())
+                .name(buildConfiguration.getName())
+                .buildScript(buildConfiguration.getBuildScript())
+                .creationTime(buildConfiguration.getCreationTime())
+                .modificationTime(buildConfiguration.getModificationTime())
+                .project(buildConfiguration.getProject())
+                .environment(buildConfiguration.getEnvironment())
+                .parameters(buildConfiguration.getParameters())
+                .scmRepository(scmClient.getSpecific(repositoryConfiguration2Id))
+                .buildType(buildConfiguration.getBuildType())
+                .build();
+        bcClient.update(updateBC.getId(), updateBC);
+        BuildConfiguration updatedBC = bcClient.getSpecific(updateBC.getId());
+
+        // then
+        assertThat(updateBC.getScmRepository().getId()).isEqualTo(repositoryConfiguration2Id);
+    }
+
+    @Test
+    public void shouldFailToCreateNewBuildConfigurationBecauseIdIsNotNull() throws ClientException {
+        BuildConfigurationClient client = new BuildConfigurationClient(RestClientConfiguration.asUser());
+
+        BuildConfiguration bc = client.getSpecific(configurationId);
+
+        assertThatThrownBy(() -> client.createNew(bc)).hasCauseInstanceOf(BadRequestException.class);
```

Comment:
You are traing to make a copy of existing? Are you sure it's failing because of non-null id and not for example existing name?